packages/0x.js/src/contract_wrappers/generated/base_contract.ts

```diff
@@ -1,11 +1,45 @@
-import {TxData, TxDataPayable} from '@0xproject/types';
+import { TxData, TxDataPayable } from '@0xproject/types';
+import { Web3Wrapper } from '@0xproject/web3-wrapper';
+import * as ethersContracts from 'ethers-contracts';
 import * as _ from 'lodash';
 import * as Web3 from 'web3';
 
 export class BaseContract {
-    protected _web3ContractInstance: Web3.ContractInstance;
-    protected _defaults: Partial<TxData>;
-    protected async _applyDefaultsToTxDataAsync<T extends TxData|TxDataPayable>(
+    protected _ethersInterface: ethersContracts.Interface;
+    protected _web3Wrapper: Web3Wrapper;
+    public abi: Web3.ContractAbi;
+    public address: string;
+    protected static _transformABIData(
+        abis: Web3.DataItem[],
+        values: any[],
+        transformation: (type: string, value: any) => any,
+    ): any {
+        return _.map(values, (value: any, i: number) =>
+            BaseContract._transformTypedData(abis[i].type, value, transformation),
+        );
+    }
+    protected static _lowercaseAddress(type: string, value: string): string {
+        return type === 'address' ? value.toLowerCase() : value;
+    }
+    protected static _bigNumberToString(type: string, value: string): string {
+        return _.isObject(value) && (value as any).isBigNumber ? value.toString() : value;
```

Comment:
Can we move the `isBigNumber` logic into a utility? Esp. since we are planning to upgrade BigNumber, it would be great if this logic lived in a single place.

Reply:
Was this addressed @LogvinovLeon